doc/design/volumes/nfsvolumestores.md

```diff
@@ -0,0 +1,106 @@
+
+Some psuedo code.
+```
+func VolumeCreate() {
+// volPath := vicVolumePath(nameOfVolume)
+// mkdir volPath
+// return volPath
+}
+```
+#### VolumeDestroy
+Likewise destroying the volume is simply removing the volume's top level directory.
+```
+func VolumeDestroy() {
+// volPath := vicVolumePath(nameOfVolume)
+// rm -rf volPath
+// return $?
+}
+```
+
+#### VolumeList
+Listing the volumes is just listing the diretories at the top of the volume store location
+```
+func VolumesList() {
+// return ls -l vicVolumePath(.)
+}
+```
+
+### Testing
+
+#### Functional
+
+ 1. Create a VCH with an NFS backed `VolumeStore`, create a volume on the `VolumeStore`, create 2 containers with the volume attached, touch a file from the first container, verify it exists on the 2nd.  Destroy the 2nd container, attempt to destroy the volume and expect a failure.  Poweroff the first container, reattempt destroy of the volume, it should fail.  Then destroy the container and destroy the volume. 
+ 2. Create a VCH with a nonreachable NFS backed `VolumeStore`.  Creation of the volume should return an error.
+ 
+#### Unit
+
+Whether the `VolumeStore` implementation uses the local VCH to mount the NFS or uses a client library to manipulate the target, the Storer implementation should sit in front of an interface which can be mocked.  The mock should write to the local filesystem so the storer interface can be tested end to end without requiring an NFS server.
+
+### Open questions
+ 1. Should we allow the default volumestore to be NFS backed?
+    * Answer:  I don't see a reason why we can't support this.
+ 1. Is there any mechanism by which we can indicate available space in the volumestore? Is this necessary data for functional usage.
+    * Answer: See Non-requirement 3
+ 1. Should we allow for read-only volume store? - e.g. publishing datasets for consumption
+    * Answer: Needs investigation.  What is RO here (the target or the directory) and what would the container user want to see or expect when such a target was used?
+ 1. Failure handling;  what do we do if a mount is unavailable, does the container go down?
+    * Answer:  Needs investigation.  We're relying on the kernel nfs client in the container to handle failures to the target.  There is little we can do during run-time, but we can check availability during container create at a minimum.
+ 1. NFS target mount options-  How do you pass a `uid`/`gid` or `nosquash` mapping?  How do you map `uid`/`gid` to the container at all?
+    * Answer:  Requirement and usage needs to be thought through at a higher level.  Mapping of users into containers and mapping of credentials into containers need to be solved in the system as a whole.
```

Comment:
iirc it's possible to pass raw mount options through at the moment when you specify `-v source:dest,opts`, but this is something that should really be hidden from the container user.